BrainPortal/app/controllers/nh_projects_controller.rb

```diff
@@ -84,12 +87,139 @@ def update #:nodoc:
 
   def show #:nodoc:
     @nh_project = find_nh_project(current_user, params[:id])
+    @can_add_license = @nh_project.creator_id == current_user.id
   end
 
   def files #:nodoc:
     @nh_project = find_nh_project(current_user, params[:id])
     @files      = @nh_project.userfiles
   end
 
+  def new_license #:nodoc:
+    @nh_project = find_nh_project(current_user, params[:id] || params[:nh_project_id], false)
+
+    unless @nh_project.creator_id == current_user.id
+      cb_error "Only owner can set licensing", :redirect  => :neurohub
+    end
+  end
+
+  def add_license #:nodoc:
+    content = params[:meta][:license_description]
+    @nh_project = find_nh_project(current_user, params[:id] || params[:nh_project_id], false)
+    cb_error 'Presently empty licenses are not allowed' if content.blank?
+    unless @nh_project.creator_id == current_user.id
+      cb_error "Only owner can set licensing", :redirect  => :neurohub
```

Comment:
You can use the same structure as before:
> cb_error ... if ...

Reply:
ok will do. Do you think it is make sense to move check into a helper or method for more dry code

Reply:
It's fine for me here.

Reply:
@natacha-beck are you fine with moving this check out to model or helper, or prefer keep it here

Reply:
Here is a good place for me.

Reply:
I d rather move out but ok, kept in in place